www/source/2017-07-14-Habitat-mysql-cluster.html.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Making a Leader/Follower MySql Cluster with Habitat!
+date: 2017-07-14
+author: Nell Shamrell-Harrington
+tags: supervisor, packaging
+category: Supervisor
+classes: body-article
+---
+
+One of the most compelling pieces of Habitat is using the supervisor to create self-organizing and self-healing topologies.  Today we will, step by step, create a MySQL Leader/Follower cluster using Habitat.
+
+## Setting up your infrastructure
+
+Create three virtual machines on the cloud provider of your choice (when creating this post, I used AWS EC2).
+
+Why virtual machines and not containers?  You certainly could create a cluster in containers, but I strongly prefer to use containers only for stateless portions of an application.  A database is stateful part of an application and should be in a long lasting form of infrastructure, rather than an ephemeral one.
+
+When you set up your virtual machines, make sure that these ports are open.
+
+* 22 (ssh)
+* 9631 (tcp)
+* 9631 (udp)
```

Comment:
We don't use 9631 UDP, just TCP. You only need to open this if you want to be able to see the http gateway of the Supervisor